ex2/sawada/main.py

```diff
@@ -0,0 +1,249 @@
+import argparse
+
+import numpy as np
+import matplotlib.pyplot as plt
+from mpl_toolkits.axes_grid1 import make_axes_locatable
+import librosa
+import librosa.display
+import scipy.io.wavfile
+
+
+def _create_base_filter(cutoff_frequency, nperseg=512, sr=16000):
+    """filter creation base function
+    
+    This function returns the basic LPF for creating filter, wrapped and used.
+
+    Args:
+        cutoff_frequency (int): cutoff frequency
+        nperseg (int, optional): samples in stft segment. Defaults to 512.
+        sr (int, optional): sampling rate. Defaults to 16000.
+        
+    Returns:
+        ndarray: filter
+    """
+    
+    regularized_cutoff_frequency = 2 * np.pi * cutoff_frequency / sr
+    t = np.arange(nperseg)
+    
+    # create filter
+    filter = 2 * regularized_cutoff_frequency \
+        * np.sinc(regularized_cutoff_frequency / np.pi * (t - nperseg // 2)) \
+        * np.hamming(nperseg)
+    return filter
+
+
+def create_lpf(cutoff_frequency, fft_size=512, sr=16000):
+    """low-pass filter
+
+    Args:
+        cutoff_frequency (int): cutoff frequency
+        fft_size (int, optional): samples in stft segment. Defaults to 512.
+        sr (int, optional): sampling rate. Defaults to 16000.
+
+    Returns:
+        ndarray: low-pass filter
+    """
+    return _create_base_filter(cutoff_frequency, fft_size, sr)
+
+
+def create_hpf(cutoff_frequency, fft_size=512, sr=16000):
+    """high-pass filter
+
+    Args:
+        cutoff_frequency (int): cutoff frequency
+        fft_size (int, optional): samples in stft segment. Defaults to 512.
+        sr (int, optional): sampling rate. Defaults to 16000.
+
+    Returns:
+        ndarray: high-pass filter
+    """
+    return _create_base_filter(sr / (2 * np.pi)) \
+        - _create_base_filter(cutoff_frequency, fft_size, sr)
+
+
+def apply_filter(audio, filter):
+    """apply filter to audio
+
+    Args:
+        audio (ndarray): audio array
+        filter (ndarray): filter
```

Comment:
追加でもう一点、 `numpy.ndarray` の関数引数にコメントつけるときは、配列の次元や各軸の意味（あるいはサイズ）が分かるようにしておくと後で幸せになれます。以下例です。
```suggestion
    Args:
        audio (ndarray, axis=(time,)): audio array
        filter (ndarray, shape=(filter_length,)): filter
```
axis で書くか shape で書くかは好みですがどっちかに統一した方がいいのは間違いないです。